File: PascalTriangle.py
# Pascal Triangle
# - Generate Pascal's Triangle up to n rows.
# - Each number is the sum of the two numbers directly above it.
# - Constraints:
#   - 1 <= numRows <= 30

from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = []
        for i in range(numRows):
            row = [1] * (i + 1)
            for j in range(1, i):
                row[j] = triangle[i-1][j-1] + triangle[i-1][j]
            triangle.append(row)
        return triangle

File: PascalTriangle.cs
// Pascal Triangle
// - Generate Pascal's Triangle up to n rows.
// - Each number is the sum of the two numbers directly above it.

using System;
using System.Collections.Generic;

class Solution
{
    public List<List<int>> Generate(int numRows)
    {
        var triangle = new List<List<int>>();

        for (int i = 0; i < numRows; i++)
        {
            var row = new List<int>(new int[i + 1]);
            for (int k = 0; k <= i; k++)
                row[k] = 1;

            for (int j = 1; j < i; j++)
            {
                row[j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }

            triangle.Add(row);
        }

        return triangle;
    }
}
